src/editor/nodes/ParticleNode.js

```diff
@@ -0,0 +1,416 @@
+import THREE from "../../vendor/three";
+import EditorNodeMixin from "./EditorNodeMixin";
+import eventToMessage from "../utils/eventToMessage";
+import defaultParticleUrl from "../../assets/dot.png";
+
+let defaultParticleSprite = null;
+
+const vertexShader = `
+      precision highp float;
+
+      #define BASE_PARTICLE_SIZE 300.0
+
+      uniform mat4 modelViewMatrix;
+      uniform mat4 projectionMatrix;
+
+
+      attribute vec4 color;
+      attribute vec4 position;
+      attribute float customAngle;
+
+      varying float vAngle;
+      varying vec4 vColor;
+      varying float vPosZ;
+
+
+      
+
+			void main() {
+        vPosZ = position.z;
+        vColor = color;
+        vAngle = customAngle;
+    
+				vec4 mvPosition = modelViewMatrix * vec4( position.xyz, 1.0 );
+				gl_PointSize = position.w * ( BASE_PARTICLE_SIZE / -mvPosition.z );
+				gl_Position = projectionMatrix * mvPosition;
+      }
+      `;
+
+const fragmentShader = `
+      precision highp float;
+
+      uniform sampler2D texture;
+
+      varying float vAngle;
+      varying vec4 vColor;
+      varying float vPosZ;
+      
+
+      void main() {
+       
+        
+        float c = cos(vAngle);
+        float s = sin(vAngle);
+        
+        gl_FragColor = vColor; 
+
+	      vec2 rotatedUV = vec2(c * (gl_PointCoord.x - 0.5) + s * (gl_PointCoord.y - 0.5) + 0.5, 
+	      c * (gl_PointCoord.y - 0.5) - s * (gl_PointCoord.x - 0.5) + 0.5);  // rotate UV coordinates to rotate texture
+    	  vec4 rotatedTexture = texture2D( texture,  rotatedUV );
+        gl_FragColor = gl_FragColor * rotatedTexture;
+        //gl_FragColor = gl_FragColor * texture2D( texture, gl_PointCoord );
+        
+        if ( vPosZ < -0.1) discard;
+        
+				
+			}
+  `;
+
+export default class ParticleNode extends EditorNodeMixin(THREE.Points) {
```

Comment:
Maybe we should rename this to `ParticleEmitterNode` and update the UI to say `Particle Emitter`?